### PR TITLE
implemented TapUp within InkResponse and InkWell

### PR DESCRIPTION
**Pull request by gslender**
implemented the missing TapUp event within InkResponse and InkWell

Issue #93683 identified this as missing 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/ink_well.dart [hunk 1]
```diff
@@ -337,6 +338,9 @@ class InkResponse extends StatelessWidget {
   /// Called when the user taps down this part of the material.
   final GestureTapDownCallback? onTapDown;
 
+  /// Called when the user releases and before [onTap].
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/material/ink_well.dart:341]:
```suggestion
  /// Called when the user taps this part of the material and releases the tap,
  /// invoked before [onTap].
```

**Review by dkwingsmt — COMMENTED** [review]
LGTM except for nit documentation. Also needs a 2nd review.

**Review by werainkhatri — CHANGES_REQUESTED** [review]
You may need to create a new issue for this PR and add it to the description like **Fixes #ISSUE_NUMBER**.

File: packages/flutter/lib/src/material/ink_well.dart [hunk 2]
```diff
@@ -337,6 +338,10 @@ class InkResponse extends StatelessWidget {
   /// Called when the user taps down this part of the material.
   final GestureTapDownCallback? onTapDown;
 
+  /// Called when the user releases a tap that was started on this part of the
```
Comment (werainkhatri) [inline packages/flutter/lib/src/material/ink_well.dart:341]: Just a minor analysis fix.

```suggestion
  /// Called when the user releases a tap that was started on this part of the
```

**Review by werainkhatri — APPROVED** [review]
LGTM. Thank you for your contribution.

**Pull request merged**